Wrong cgen with `go fn(){}()`

**V version:** 
0.1.29 664c26a

**OS:** 
Endeavouros / Arch Linux in VirtualBox VM on Windows 10 Pro
**What did you do?**

// hello.v
fn main() {
	go fn() {
		for {
			println("Hello V!")
		}
	}()
}

v run hello.v

**What did you expect to see?**
Hello V!
Hello V!
Hello V!
Hello V!
Hello V!
....... infinite loop
 
**What did you see instead?**
builder error: 
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

I get:
```
C:\Users\CZASTK~1\AppData\Local\Temp\v\goerr.tmp.c: In function '_thread_wrapper':
C:\Users\CZASTK~1\AppData\Local\Temp\v\goerr.tmp.c:1803:3: error: expected expression before ')' token
  ();
   ^
```

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -4352,6 +4352,10 @@ fn (mut g Gen) go_stmt(node ast.GoStmt) {
 	if expr.is_method {
 		receiver_sym := g.table.get_type_symbol(expr.receiver_type)
 		name = receiver_sym.name + '_' + name
+	} else if expr.left is ast.AnonFn as anon_fn {
+		g.gen_anon_fn_decl(anon_fn)
+		fsym := g.table.get_type_symbol(anon_fn.typ)
+		name = fsym.name
 	}
 	name = util.no_dots(name)
 	g.writeln('// go')

--- a/vlib/v/tests/anon_fn_test.v
+++ b/vlib/v/tests/anon_fn_test.v
@@ -1,0 +1,9 @@
+import sync
+
+fn test_go_anon_fn() {
+	mut wg := sync.new_waitgroup()
+	go fn (mut wg sync.WaitGroup) {
+		wg.done()
+	}(mut wg)
+	wg.wait()
+}

